### PR TITLE
[framework]Implement mux-cable flap counter in mux_simulator_server

### PR DESCRIPTION
**Pull request by bingwang-ms**
Signed-off-by: bingwang <bingwang@microsoft.com>

### Description of PR

Summary:
Fixes # (issue)
This PR implements mux_cable flap counter in ```mux_simulator_server```.
The flap counter can be retrieved and cleared by HTTP request.

### Type of change

- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
This PR is to implement mux-cable flap counter in mux_simulator.

#### How did you do it?
Please see code.

#### How did you verify/test it?
Verified by a demo test script.
```
def test_flap_counter(simulator_flap_counter, simulator_flap_counters, toggle_simulator_port_to_upper_tor, toggle_simulator_port_to_lower_tor, simulator_clear_flap_counter, simulator_clear_flap_counters):
    counter1 = simulator_flap_counter('Ethernet0')
    assert(counter1 == 0)
    toggle_simulator_port_to_upper_tor('Ethernet0')
    counter2 = simulator_flap_counter('Ethernet0')
    assert(counter2 == counter1 + 1)
    simulator_clear_flap_counter('Ethernet0')
    counter3 = simulator_flap_counter('Ethernet0')
    assert(counter3 == 0)
```
#### Any platform specific information?
dualtor specific.

### CODE REVIEW

File: ansible/roles/vm_set/files/mux_simulator.py [hunk 1]
```diff
@@ -28,6 +28,50 @@
...
+    @summary: Clear the flap counter for a certain mbr
+    @param vm_set: A str, vm_set name
+    @param port_index: An integer, the port_index
+    @return: None
+    """
+    mbr_name = "mbr-{}-{}".format(vm_set, port_index)
+    flap_counter[mbr_name] = 0
+
+
+def fill_flap_counter():
```
Comment (theasianpianist) [inline ansible/roles/vm_set/files/mux_simulator.py:67]: A better name might be `init_flap_counter`? 'Fill' seems a little ambiguous

File: tests/common/dualtor/mux_simulator_control.py [hunk 2]
```diff
@@ -322,3 +327,51 @@ def _drop_helper(interface_name):
...
+@pytest.fixture
+def simulator_flap_counter(url):
+    """
+    A function level fixture to retrieve mux simulator flap counter for a given interface
+    """
+    def _simulator_flap_counter(interface_name):
+        server_url = url(interface_name, FLAP_COUNTER)
+        counter = _get(server_url)
+        assert(counter and len(counter) == 1)
+        return counter.values()[0]
```
Comment (theasianpianist) [inline tests/common/dualtor/mux_simulator_control.py:340]: To avoid a `TypeError`

```suggestion
        return list(counter.values())[0]
```

Comment (bingwang-ms) [reply tests/common/dualtor/mux_simulator_control.py]: Thanks. Updated.

File: ansible/roles/vm_set/files/mux_simulator.py (hunk 1 above)
Comment (bingwang-ms) [reply ansible/roles/vm_set/files/mux_simulator.py]: Good suggestion. Updated. Thanks

**Review by theasianpianist — APPROVED** [review]

**Pull request merged**